src/main/kotlin/no/nav/cv/eures/util/AvroHandling.kt

```diff
@@ -11,43 +11,36 @@ import org.slf4j.LoggerFactory
 class AvroHandling {
 }
 
-// Julian:
-// This is definitely not the best solution, but unfortunately
-// it's the only one I see for handling both avro versions without
-// ending up in situations where we re-index things and get errors
-// until we finish processing the first part of the topic.
+fun List<Byte>.toHex(): String = joinToString(separator = "") { eachByte -> "%02x".format(eachByte) }
+
 fun ByteArray.toMelding(aktorId: String): Melding {
     val log = LoggerFactory.getLogger(AvroHandling::class.java)
 
-    // NOTE: The newest AVRO version prefixes 6 bytes instead of 4
+    val avroPrefixByteSize = 5 // Byte 0 is magic byte, bytes 1-4 is avro schema version - BEWARE OF VERSION
 
     return try {
-        readDatum(aktorId = aktorId, avroPrefixByteSize = 5)
-    } catch (e: Exception) {
-        log.warn("Failed to decode kafka message for $aktorId in the first try, retrying. Size: $size", e)
-        readDatum(aktorId = aktorId, avroPrefixByteSize = 7)
-    } catch (e: Exception) {
-        log.error("Klarte ikke decode kafka melding for $aktorId. Size: $size", e)
-        throw(e)
-    }
-}
+        if(size < avroPrefixByteSize)
+            throw Exception("Trying to decode a message of only $size bytes, with a prefix of $avroPrefixByteSize")
 
-private fun ByteArray.readDatum(aktorId: String, avroPrefixByteSize: Int): Melding {
-    val log = LoggerFactory.getLogger(AvroHandling::class.java)
+        val compatibleAvroVersion = "Ox00000000"
+        val messageAvroVersion = slice(1..4).toHex()
 
-    if(size < avroPrefixByteSize)
-        throw Exception("Trying to decode a message of only $size bytes, with a prefix of $avroPrefixByteSize")
+        if(!messageAvroVersion.equals(compatibleAvroVersion)) {
+            val e = Exception("Trying to decode a message with the wrong avrfo version! Version is $messageAvroVersion, but should be $compatibleAvroVersion")
+            log.error("Unable to decode avro message", e)
+            throw e
```

Comment:
Kunne vi kanskje lagt til `panic` funksjonalitet på dette?  Hvis vi ikke kan dekode en avro melding, så slutter applikasjonen å konsumere kafka fullt, hva tenker dere?

Reply:
Tja, tjo, den vil jo logge masse errors og aldri gå videre, så slack kanalen vår vil få alerts. Tenker det blir litt innviklet og error prone å ha `panic` mode for kafka konsumeren. Men om dere andre stemmer for gjør vi det!

Reply:
Tenkte det mer som at vi da sikrer oss at vi ikke kan lagre noe i databasen som er av feil type, men vi vil nok merke ting veldig fort når vi først får errors på slack som du sier, så kanskje unødvendig.

Reply:
Vi kan tenke på det et par dager :)